VisualizationBase/src/nodes/InfoNode.cpp

```diff
@@ -34,7 +34,7 @@ namespace Visualization {
 
 DEFINE_TYPE_ID_DERIVED(InfoNode, "InfoNode", )
 
-QList<InfoNode*> InfoNode::allInfoNodes;
+QHash<QString, InfoNode*> InfoNode::allInfoNodes;
```

Comment:
Given that your key is just an integer, using a `QList` is just fine here as you can simply do `allInfoNodes[key]`.